compiler/ast/typesrenderer.nim

```diff
@@ -154,6 +154,24 @@ proc rangeToStr(n: PNode): string =
   assert(n.kind == nkRange)
   result = valueToString(n[0]) & ".." & valueToString(n[1])
 
+proc genericParamsToStr*(n: PNode): string =
```

Comment:
`renderer.renderTree` already supports rendering `nkGenericParams` trees -- that should be used instead.